### PR TITLE
Integration with gh-pages

### PR DESCRIPTION
**Pull request by marquiz**
This PR enables automated generation of html-based documentation served by [GitHub Pages](https://pages.github.com/).

It sets up a fresh, virtually empty, [Jekyll](https://jekyllrb.com/)-based site under `docs/`, using [jekyll-rdt theme](https://github.com/rundocs/jekyll-rtd-theme). It adds makefile rules for generating and serving the documentation locally. A new `scripts/github/update-gh-pages.sh` script is introduced - designed to serve as an integration point to github actions. Finally, the PR introduces a github workflow for updating the gh-pages branch and a test-infra script for verifying buildability of the site (gh-pages).

The layout of the `gh-pages` branch is supposed to look like this:
```
.
├── index.html
├── master/
├── v0.6/
├── v0.x/
...
```
I.e. every major release (branch) would have it's own subdir. Whenever a release tag (in the format `vX.Y.Z`) is pushed or release branch (`release-X.Y`) updated, the corresponding documentation subdir in gh-pages will be re-built. `index.html` will just redirect to the latest (stable when we have one) version.

This PR implements the plumbing described above. Content is out-of-scope of this PR - it will be addressed when the pipeline works.

Output (manually generated by running `scripts/test-infra/gh-pages.sh`) can be viewed from my personal repo:
https://marquiz.github.io/node-feature-discovery/devel/gh-pages-debug

### CODE REVIEW

**Comment by marquiz** [comment]
I think this is now ready for review. Gh-pages auto-generation is working in my personal repo

File: Makefile [hunk 1]
```diff
@@ -7,6 +7,16 @@ GO_FMT ?= gofmt
 IMAGE_BUILD_CMD ?= docker build
 IMAGE_BUILD_EXTRA_OPTS ?=
 IMAGE_PUSH_CMD ?= docker push
+IMAGE_RUN_CMD ?= docker run
```
Comment (ArangoGutierrez) [inline Makefile:10]: CONTAINER_RUN...?

File: scripts/github/update-gh-pages.sh [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+site_subdir="$1"
+
+# Check that no extra args were provided
+if [ $# -gt 1 ]; then
+    echo "ERROR: extra positional arguments: $@"
+    usage
+    exit 1
+fi
+
+
```
Comment (ArangoGutierrez) [inline scripts/github/update-gh-pages.sh:45]:
```suggestion
```

File: scripts/github/update-gh-pages.sh [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+fi
+
+# Default to 'master' if no subdir was given and we couldn't parse
+# it
+export SITE_SUBDIR=${site_subdir:-master}
+echo "Updating site subdir: '$SITE_SUBDIR'"
+
+make site-build
+
+
```
Comment (ArangoGutierrez) [inline scripts/github/update-gh-pages.sh:96]:
```suggestion
```

**Review by ArangoGutierrez — COMMENTED** [review]
looking good

File: Makefile (hunk 1 above)
Comment (marquiz) [reply Makefile]: Yes, more appropriate. Changed

File: scripts/github/update-gh-pages.sh (hunk 2 above)
Comment (marquiz) [reply scripts/github/update-gh-pages.sh]: Removed

File: scripts/github/update-gh-pages.sh (hunk 3 above)
Comment (marquiz) [reply scripts/github/update-gh-pages.sh]: Removed

**Comment by marquiz** [comment]
Please don't force squashing. I don't want to lose all the commit messages. I now manually squashed the PR into a more compact patchset.
/remove-label tide/merge-method-squash

**Review by ArangoGutierrez — APPROVED** [review]
My requests have been addressed, this patch is good to go, from my point of view
/lgtm

**Pull request merged**